### PR TITLE
[protobuf-cpp] Initial plan and tests

### PR DESCRIPTION
**Pull request by MindNumbing**
Signed-off-by: Steven Marshall <SMarshall@chef.io>

### CODE REVIEW

File: controls/protobuf-cpp_works.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  impact 1.0
+  title 'Ensure protobuf-cpp binary is working as expected'
+  desc '
+  We first check that the protobuf-cpp binary we expect is present and then run version checks on both to verify that it is excecutable.
+  '
+
+  hab_pkg_path = command("hab pkg path #{plan_ident}")
+  describe hab_pkg_path do
+    its('exit_status') { should eq 0 }
+    its('stdout') { should_not be_empty }
```
Comment (gavindidrichsen) [inline controls/protobuf-cpp_works.rb:21]: Extra sterr check, its('stderr') { should be_empty }

File: controls/protobuf-cpp_works.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  target_dir = File.join(hab_pkg_path.stdout.strip, base_dir)
+
+  protoc_exists = command("ls #{File.join(target_dir, "protoc")}")
+  describe protoc_exists do
+    its('stdout') { should match /protoc/ }
+    its('stderr') { should be_empty }
+    its('exit_status') { should eq 0 }
+  end
+
+  protoc_works = command("/bin/protoc --version")
```
Comment (gavindidrichsen) [inline controls/protobuf-cpp_works.rb]: Could you use the full /hab/pkgs/core/.../protoc path here instead to ensure that the actual installation is getting tested.  You could reuse the File.join above, I think.

**Review by gavindidrichsen — COMMENTED** [review]
Just a couple minor updates, but otherwise looks good

**Review by skpaterson — APPROVED** [review]
Thanks @MindNumbing - looks like all comments are addressed.

**Pull request merged**